security-malware-firewall.php

```diff
@@ -2542,3 +2545,63 @@ function spbc_private_list_add( $ip ) {
 	}
 	return false;
 }
+
+/**
+ * Checks if there is a difference between sent and received files lists. If so, resend files. Files sending retries restricted by 10 minutes delay by defaults.
+ *
+ * @param $delay - delay in minutes. Default is 10. If user calls sync failed files won't send if they are already have been sent in delay.
+ * @global State $spbc
+ * @global wpdb $wpdb
+ * @return void
+ */
+function spbc_resend_failed_files_for_analysis($delay = 10){
+    global $spbc, $wpdb;
+    //reading DB for NEW files
+    $sent_files_data = $wpdb->get_results(
+        'SELECT path, fast_hash, last_sent'
+        . ' FROM '.SPBC_TBL_SCAN_FILES
+        . ' WHERE analysis_status = "NEW"',
+        ARRAY_A
+    );
+    //extract paths
+    $sent_paths = array_column($sent_files_data,'path');
+
+    //get files received from CT backend
+    $received_files_data = SpbcAPI::method__security_mscan_status(
+        $spbc->settings['spbc_key'],
+        $sent_paths
+    );
+    //stop if method return error
+    if ( ! empty($result['error']) ) {
+        return;
+    }
+    //extract paths
+    $received_paths = array_column($received_files_data,'file_path');
```

Comment:
what if the array is empty or there is an error when adding the base, there will be no warning or notification

Reply:
Fixed.